platform/ios/app/MBXViewController.m

```diff
@@ -2416,8 +2371,33 @@ - (void)updateHUD {
 
 - (void)mapViewDidFinishRenderingFrame:(MGLMapView *)mapView fullyRendered:(BOOL)fullyRendered {
     if (self.frameTimeGraphEnabled) {
+        NSLog(@"⏰ FRAMETIME: %@", mapView.frameTime);
         [self.frameTimeGraphView updatePathWithFrameDuration:mapView.frameTime];
     }
 }
 
+- (void)saveCurrentMapState:(__unused NSNotification *)notification {
+
+    // The following properties can change after the view loads so we need to save their
+    // state before exiting the view controller.
+    self.currentState.camera = self.mapView.camera;
+    self.currentState.showsUserLocation = self.mapView.showsUserLocation;
+    self.currentState.userTrackingMode = self.mapView.userTrackingMode;
+    self.currentState.showsUserHeadingIndicator = self.mapView.showsUserHeadingIndicator;
+    self.currentState.showsMapScale = self.mapView.showsScale;
+    self.currentState.showsZoomLevelOrnament = self.zoomLevelOrnamentEnabled;
+    self.currentState.showsTimeFrameGraph = self.frameTimeGraphEnabled;
+    self.currentState.debugMask = self.mapView.debugMask;
+    self.currentState.debugLoggingEnabled = self.debugLoggingEnabled;
+
+    [[MBXStateManager sharedManager] saveState:self.currentState];
+
+}
+
+- (void)restoreMapState:(__unused NSNotification *)notification {
+    MBXState *currentState = [MBXStateManager sharedManager].currentState;
```

Comment:
The restoration is inconsistent. If you prefer to store the properties in the view controller then you should do something like:
```
self.debugLoggingEnabled = self.currentState.debugLoggingEnabled;
self.mapView.camera = self.currentState.camera;
...
```
But storing some properties in the view controller and other in an `MBXState` object will cause state bugs.